### DISCUSSION TITLE
Colorized keymap zones

### DISCUSSION THREAD

**Discussion by alexchexes** [monkeytypegame/monkeytype#5931] [Ideas]
I wonder why the most popular touch-typing tool doesn't have any kind of keymap color guides, which are a must-have for beginners or near-beginners.

I've created a simple userscript for myself that generates CSS to color and highlight the keymap in one of three styles of your choice ([here's the link](https://gist.github.com/alexchexes/498c069fa994b7d76d8bcc1f7c144629)).
Check out how it looks in these videos:

https://github.com/user-attachments/assets/60e06ab6-b3fe-47db-93db-dc65d0d669c1

https://github.com/user-attachments/assets/31ef43b4-b20b-4c90-ab3e-1086356f4854

https://github.com/user-attachments/assets/9c59d2ae-6d33-43af-9a50-865337c822bd

I’d love to see this implemented as a standard feature.

A couple of notes:

1. The color choices are deliberate. Although other people might be used to differently colored keys in other tools (to address this, it would be useful to add the ability to customize colors in the settings), the current selection works well as mnemonic aids:
- Pink (lavender) for the pinky finger. This works for English speakers, but for others, pink may also stereotypically represent a "girly" color and might be associated with cute, small things. Stupid as it sounds, this mnemonic could work well.
- Gold (orange/yellow) for the ring finger, since rings are often gold. Works at least for Western culture.
- Red for the middle finger because you probably shouldn’t show your middle finger to people.
- The left and right index fingers are the brightest (lime and cyan, or green and blue), as these are the most convenient fingers to use. For many, green and blue are the most appealing colors, and they follow the order of RGB.
2. My solution is based on this Stylus style: https://userstyles.world/style/17547/monkeytype-keymap-color
If my version is used, half the credit should go to the author of that style, as they saved me the time I would have spent matching class names with fingers.

**Comment by Miodec** [comment]
This won't really work because of themes. We cant have set colors that don't change with themes because they might blend it or become straight up invisible on some themes.

**Comment by alexchexes** [comment]
@Miodec It’s possible to create a basic colored-keymap style where the letters remain visible on any theme, and then allow theme creators to update their themes with new colors that override it. Here's an example I tested on 15 different themes—ranging from very dark to very light or with saturated background colors.

Of course, this kind of "basic" color style wouldn’t be the most beautiful, but it works. If theme creators want to, they can adjust the colors to better fit their theme. Anyway, it would likely be a disabled feature by default.

![chrome_2024-10-07--12-20-02--293](https://github.com/user-attachments/assets/b03d4470-8f21-423d-87e7-2ac27d324fe3)
![chrome_2024-10-07--12-20-16--028](https://github.com/user-attachments/assets/f114e5bd-d4d2-41b9-a168-8153b7889be2)
![chrome_2024-10-07--12-20-50--064](https://github.com/user-attachments/assets/34375d74-d230-41ad-a459-92b9e10951a3)
![chrome_2024-10-07--12-22-10--896](https://github.com/user-attachments/assets/d8238f4b-a1fc-4725-aab8-afc3e16f0f59)
![chrome_2024-10-07--12-22-21--029](https://github.com/user-attachments/assets/9237fc8b-679e-45e1-bd79-58218bf254ed)
![chrome_2024-10-07--12-17-21--418](https://github.com/user-attachments/assets/b374375f-7613-4a2a-84b0-fff86b28dd4e)
![chrome_2024-10-07--12-17-40--836](https://github.com/user-attachments/assets/99540dbc-0dd7-428c-bffa-58019dbd9a0c)
![chrome_2024-10-07--12-17-59--474](https://github.com/user-attachments/assets/3fe806d7-35a7-4b52-b395-34cdb6ffcd1b)
![chrome_2024-10-07--12-18-12--888](https://github.com/user-attachments/assets/735a5acf-a7a2-4061-828d-48c52c44dc7a)
![chrome_2024-10-07--12-18-26--681](https://github.com/user-attachments/assets/26bd4f5d-6812-4ee3-b9f9-8160bfdedd58)
![chrome_2024-10-07--12-18-40--378](https://github.com/user-attachments/assets/4ed8e422-334c-4c1f-b31a-22ac2119a4c0)
![chrome_2024-10-07--12-19-02--042](https://github.com/user-attachments/assets/aa0e4c7f-dd36-4cba-a82e-a031a872d439)
![chrome_2024-10-07--12-19-19--070](https://github.com/user-attachments/assets/9d91bf19-3457-4e67-baf6-903a60aebf6f)
![chrome_2024-10-07--12-19-30--581](https://github.com/user-attachments/assets/2e055521-096a-4465-8dc8-ade27f61ee7d)
![chrome_2024-10-07--12-19-44--052](https://github.com/user-attachments/assets/171d67eb-eb12-48e4-9df4-f68f06cfea98)
